src/lib/application.rs

```diff
@@ -0,0 +1,23 @@
+use crate::sensor::SensESPSensor;
+
+pub struct Application {
+    sensors: Vec<Box<dyn SensESPSensor>>,
```

Comment:
the `dyn` format is what you're really going to look for when you start wanting OOP generics.  This allows for a vector (an array-like type) of owned objects (the `Box`) that implement a given trait, `SensESPSensor`. 

Rust is like C++, and unlike Java, in that templates create concrete classes, and other forms of generic in rust require the same concrete type, even if two different types individually satisfy the restraint.